src/Network.cpp

```diff
@@ -117,23 +117,38 @@ namespace libclient {
                 for (const auto &op: m.getOperations()) {
                     model->gameState.operations.push_back(op->clone());
                 }
-                model->gameState.state = m.getState();
+                model->gameState.handleLastClientOperation(m.getState());
                 model->gameState.isGameOver = m.getIsGameOver();
 
-                model->gameState.handleLastClientOperation();
-
-                if (state == IN_EQUIPMENTCHOICE) {
-                    // first GameStatus message
-                    onFirstGameStatus();
+                // update AIState for each operation
+                if (model->clientState.role != spy::network::RoleEnum::SPECTATOR) {
+                    auto myFaction = model->clientState.amIPlayer1() ? spy::character::FactionEnum::PLAYER1
+                                                                     : spy::character::FactionEnum::PLAYER2;
+                    for (const auto &op: m.getOperations()) {
+                        model->aiState.processOperation(op, model->gameState.state, model->gameState.settings, myFaction);
+                        model->aiState.applySureInformation(model->gameState.state, myFaction);
+                    }
                 }
 
-                model->aiState.processOperationList(m.getOperations());
-
+                // set current state and merge AIState into
+                model->gameState.state = m.getState();
                 if (model->clientState.role != spy::network::RoleEnum::SPECTATOR) {
```

Comment:
Mir ist nicht ganz klar was hier gemacht wird.
Oben hat man die ApplySureInformation nach jeder Operaton gerufen und jetzt hier nochmal..?

Reply:
Ja, weil ich `processOperation()` auf dem State machen muss, der die Operation noch nicht enthält und der State den ich geschickt bekomme ist, wenn alle Operationen in der Liste ausgeführt wurden